rmf_robot_sim_common/src/slotcar_common.cpp

```diff
@@ -341,6 +346,8 @@ void SlotcarCommon::handle_ackermann_path_request(
         Eigen::Vector2d(wp[2].x(), wp[2].y()));
 
       AckermannTrajectory& last_traj = this->ackermann_trajectory.back();
+      if (locations[i].obey_approach_speed_limit)
+        sp2.approach_speed_limit = locations[i].approach_speed_limit;
```

Comment:
`approach_speed_limit` should be check that it is larger than 0, as it looks like it is not checked further down the code path, https://github.com/open-rmf/rmf_simulation/blob/feature/speed_limits/rmf_robot_sim_common/src/utils.cpp#L32